integration-tests/src/test_loop/utils/transactions.rs

```diff
@@ -499,3 +495,164 @@ pub fn make_accounts(num_accounts: usize) -> Vec<AccountId> {
 pub fn make_account(index: usize) -> AccountId {
     format!("account{}", index).parse().unwrap()
 }
+
+/// Runs a transaction until completion.
+/// Works in a non-blocking way which allows to run multiple transactions in parallel.
+/// Is meant to use with run_until.
+pub struct TransactionRunner {
```

Comment:
noice!!

Reply:
I'm not sure if it that's the way it's supposed to be done in TestLoop, but I think it's a useful utility to have

Reply:
Yeah, I noticed the changes to making run_until take a FnMut, but I couldn't see too much of a downside other than the fact that you don't really know when the condition function is called within testloop.

Example, incrementing some counter within the condition FnMut might not be a great idea.